plugins/inputs/mesos/mesos.go

```diff
@@ -17,33 +17,57 @@ import (
 	jsonparser "github.com/influxdata/telegraf/plugins/parsers/json"
 )
 
+type Role string
```

Comment:
@sparrc:
I'm not sure I understand you correctly. You want me to create two separate types like this:
```go
type SlaveRole string
type MasterRole string
```
I did implement one type (based on string) to allow easy mapping between this constant and string (used when specifying tags). If that is the problem I could refactor it to use `iota` and get rid of custom types at all.